src/main/java/edu/cornell/kfs/sys/cache/EhcacheEventListenerForUpdatingRedis.java

```diff
@@ -0,0 +1,180 @@
+import org.kuali.kfs.sys.KFSConstants;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.data.redis.connection.DataType;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.core.Cursor;
+import org.springframework.data.redis.core.KeyScanOptions;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ScanOptions;
+
+import net.sf.ehcache.CacheException;
+import net.sf.ehcache.Ehcache;
+import net.sf.ehcache.Element;
+import net.sf.ehcache.event.CacheEventListener;
+
+public class EhcacheEventListenerForUpdatingRedis implements CacheEventListener, DisposableBean {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final int KEY_BATCH_SIZE = 50;
+
+    private final RedisTemplate<String, String> redisTemplate;
+    private final long defaultTimeToLive;
+    private final ExecutorService redisExecutor;
+
+    public EhcacheEventListenerForUpdatingRedis(RedisTemplate<String, String> redisTemplate, Long defaultTimeToLive) {
+        Objects.requireNonNull(redisTemplate, "redisTemplate cannot be null");
+        Objects.requireNonNull(defaultTimeToLive, "defaultTimeToLive cannot be null");
+        int poolSize = Runtime.getRuntime().availableProcessors();
+        this.redisTemplate = redisTemplate;
+        this.defaultTimeToLive = defaultTimeToLive;
+        this.redisExecutor = new ThreadPoolExecutor(poolSize, poolSize, 0L, TimeUnit.SECONDS,
+                new LinkedBlockingQueue<>());
+    }
+
+    @Override
+    public void notifyElementPut(Ehcache cache, Element element) throws CacheException {
+        addCacheKeyPlaceholderInRedisIfAbsent(cache, element);
+    }
+
+    @Override
+    public void notifyElementUpdated(Ehcache cache, Element element) throws CacheException {
+        addCacheKeyPlaceholderInRedisIfAbsent(cache, element);
+    }
+
+    private void addCacheKeyPlaceholderInRedisIfAbsent(Ehcache cache, Element element) {
+        String cacheName = cache.getName();
+        String localCacheKey = Objects.toString(element.getObjectKey());
+        long timeToLive = determineTimeToLiveForCache(cache);
+        
+        redisExecutor.execute(() -> {
+            String redisCacheKey = CuCompositeCacheKey.buildRedisCacheKey(cacheName, localCacheKey);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("addCacheKeyPlaceholderInRedisIfAbsent, Adding placeholder value in redis for key: "
+                        + redisCacheKey);
+            }
+            redisTemplate.opsForValue().setIfAbsent(redisCacheKey, KFSConstants.ACTIVE_INDICATOR,
+                    timeToLive, TimeUnit.SECONDS);
+        });
+    }
+
+    private long determineTimeToLiveForCache(Ehcache cache) {
+        long timeToLive = cache.getCacheConfiguration().getTimeToLiveSeconds();
+        return (timeToLive > 0) ? timeToLive : defaultTimeToLive;
```

Comment:
if the time t live is less than 0, should that be a warning or the like so we can fix the configuration?

Reply:
If the value is 0 or potentially negative in Ehcache, that means Ehcache does not specify a time to live. This code just enforces a time to live in such cases. (Note that the caches that are excluded from Redis handling, like BatchFile, will not be affected by this code.)

We can use a follow-up user story to reevaluate this setup, unless it needs to be addressed now.

Reply:
later sounds reasonable